internal/jobcontainers/mounts.go

```diff
@@ -0,0 +1,56 @@
+package jobcontainers
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+)
+
+// namedPipePath returns true if the given path is to a named pipe.
+func namedPipePath(p string) bool {
+	return strings.HasPrefix(p, `\\.\pipe\`)
+}
+
+// Strip the drive letter (if there is one) so we don't end up with "%CONTAINER_SANDBOX_MOUNT_POINT%"\C:\path\to\mount
+func stripDriveLetter(name string) string {
+	// Remove drive letter
+	if len(name) == 2 && name[1] == ':' {
+		name = "."
+	} else if len(name) > 2 && name[1] == ':' {
+		name = name[2:]
+	}
+	return name
+}
+
+// setupMounts adds the custom mounts requested in the OCI runtime spec. Mounts are a bit funny as you already have
+// access to everything on the host, so just symlink in whatever was requested to the path where the container volume
+// is mounted. At least then the mount can be accessed from a path relative to the default working directory/where the volume
+// is.
+func setupMounts(spec *specs.Spec, sandboxVolumePath string) error {
+	for _, mount := range spec.Mounts {
+		if mount.Destination == "" || mount.Source == "" {
+			return fmt.Errorf("invalid OCI spec - a mount must have both source and a destination: %+v", mount)
+		}
+
+		if namedPipePath(mount.Source) {
+			return errors.New("named pipe mounts not supported for job containers - interact with the pipe directly")
+		}
+
+		fullCtrPath := filepath.Join(sandboxVolumePath, stripDriveLetter(mount.Destination))
```

Comment:
what about dos paths like \\.\C:\Test\Foo.txt or \\?\C:\Test\Foo.txt?

Reply:
I feel like if anyone passes these as volume mount paths they should just.. not..

Reply:
what about unc paths? Do we know if there are scenarios for either? I'm specifically thinking about volume mounts or server paths

Reply:
Do we even support specifying a unc path for directory mounts for server containers? I'd imagine you'd have to map a remote share to a drive letter and provide that path.

Reply:
That seems like what we recommend for shares https://docs.microsoft.com/en-us/virtualization/windowscontainers/manage-containers/persistent-storage#configuration-steps

And if someones specifying a unc path (\\localhost\c$\my_dir) to a local file/folder then.. once again they probably shouldn't haha

Reply:
That's a great question 😅 could you look into this?

Reply:
Sure, but for alpha I don't think this is really a blocker for job containers, we just want local directory mounts to function. Although I'll still look into the above as I'm curious too